soc/arm/nxp_imx/rt/CMakeLists.txt

```diff
@@ -30,3 +30,37 @@ zephyr_linker_section_configure(
   KEEP
   PRIO 11
 )
+
+
+# When dual core builds are enabled, the following steps occur:
+# Primary core image is compiled (but not linked)
+# Secondary core image is compiled and linked
+# Secondary core image bin file is converted into an ELF with one section
+# Primary core image is linked, and places converted secondary core image bin into flash
+if (CONFIG_DUAL_CORE_MCUX)
+  if (CONFIG_PRIMARY_CORE_MCUX)
+    zephyr_linker_sources(SECTIONS sections.ld)
+    # Add the secondary core image as an external library
+    add_library(remote_core_image STATIC IMPORTED GLOBAL)
+    add_dependencies(remote_core_image dual_core_remote)
+	# Because the only build byproduct defined for the external project is
+	# KERNEL_BIN_NAME, we use this as the file we link to in order to enforce
+	# a dependency that the secondary code image be built before the primary
+	# one is linked
+    set_target_properties(remote_core_image PROPERTIES IMPORTED_LOCATION
+		${REMOTE_ZEPHYR_DIR}/${KERNEL_BIN_NAME})
+	target_link_libraries(app PUBLIC remote_core_image)
```

Comment:
This looks like an artificial dependency.

`app` is not really depending upon `remote_core_image` as far as I can tell.

Reply:
It may be the wrong thing to depend on. The goal here is to force the primary image to depend on the remote image. Is there another target I could depend on besides `app` to ensure that dependency is enforced?

The reason for that requirement is that the final output dual core image is an image intended to run on the primary core, with the second core's image linked in as a section. This image is read from flash into RAM by the primary core, which then boots the secondary core from RAM.